Avoid glob_recurse:

sorted(set('  ${PROJECT_SOURCE_DIR}/' + str(i.parent) + f'/*.{ext}' 
for ext in ['hpp', 'hxx'] for i in Path('src').glob(f'**/*.{ext}')))

--- CMakeLists.txt.orig	2020-11-26 15:56:55.000000000 +0100
+++ CMakeLists.txt	2020-12-02 10:55:59.596461813 +0100
@@ -173,9 +173,39 @@
 # ----------------------------------------------------
 # --- INCLUDE ----------------------------------------
 # ----------------------------------------------------
-FILE(GLOB_RECURSE HEADERS 
+file(GLOB HEADERS
      ${PROJECT_SOURCE_DIR}/src/*.hpp 
-     ${PROJECT_SOURCE_DIR}/src/*.hxx 
+  ${PROJECT_SOURCE_DIR}/src/algorithm/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/algorithm/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/autodiff/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/casadi/math/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/casadi/spatial/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/casadi/utils/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/cppad/algorithm/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/cppad/math/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/cppad/spatial/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/autodiff/cppad/spatial/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/autodiff/cppad/utils/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/codegen/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/container/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/core/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/math/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/math/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/multibody/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/multibody/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/multibody/joint/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/multibody/joint/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/multibody/liegroup/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/multibody/liegroup/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/multibody/visitor/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/parsers/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/parsers/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/parsers/urdf/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/parsers/urdf/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/serialization/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/spatial/*.hpp
+  ${PROJECT_SOURCE_DIR}/src/spatial/*.hxx
+  ${PROJECT_SOURCE_DIR}/src/utils/*.hpp
      ) 
 
 IF(NOT BUILD_WITH_URDF_SUPPORT)
